python/lsst/ts/wep/cwfs/DonutTemplatePhosim.py

```
@@ -0,0 +1,92 @@
+
+    def makeTemplate(self, sensorName, defocalType, imageSize, phosimTemplateDir=None):
+        """
+        Make the donut template image from phosim templates.
+        The templates should have been created using
+        `bin.src/runCreatePhosimDonutTemplates.py`.
+        See the `ts_wep` docs for more information on
+        how to generate the templates.
+
+        Parameters
+        ----------
+        sensorName : str
+            The camera detector for which we want to make a template. Should
+            be in "Rxx_Sxx" format.
+        defocalType : enum 'DefocalType'
+            The defocal state of the sensor.
+        imageSize : int
+            Size of template in pixels. The template will be a square.
+        phosimTemplateDir : str, optional
+            Specify the location where the phosim templates are stored. If None
+            then the code will look in `policy/cwfs/templateData/phosimTemplates`.
+            (The default is None)
+
+        Returns
+        -------
+        numpy.ndarray [int]
+            The donut template as a binary image.
+        """
+
+        if phosimTemplateDir is None:
+            configDir = getConfigDir()
+            phosimTemplateDir = os.path.join(
+                configDir, "cwfs", "donutTemplateData", "phosimTemplates"
+            )
+
+        if defocalType == DefocalType.Extra:
+            templateFilename = os.path.join(
+                phosimTemplateDir, "extra_template-%s.txt" % sensorName
+            )
+        else:
+            templateFilename = os.path.join(
+                phosimTemplateDir, "intra_template-%s.txt" % sensorName
+            )
+        templateArray = np.genfromtxt(templateFilename, dtype=np.int)
+
+        # Make the template the desired square shape by trimming edges of template
+        templateSize = np.shape(templateArray)
+
+        # Find the excess number of pixels in x and y direction
+        templateTrim = templateSize - np.array((imageSize, imageSize))
```

What will happen if `templateSize < np.array((imageSize, imageSize))`? Do you want to raise an error or do the padding instead?